### PR TITLE
Do not use dirty Secret Bindings

### PR DESCRIPTION
**Pull request by franpog859**
**Description**

Currently, HAP could provide a "dirty" Secret Binding for the new SKR when a tenant deprovisioned a runtime (Secret Binding is marked as "dirty") and provisioned a runtime before the clean up was performed by Subscription Cleanup Job ([HAP uses the existing, "dirty" Secret Binding](https://github.com/kyma-project/control-plane/blob/main/components/kyma-environment-broker/common/hyperscaler/account_pool.go#L112)). In such a case, all the resources would be removed by Subscription Cleanup Job as it cleans up everything from all the Secret Bindings marked "dirty"

Changes proposed in this pull request:
- HAP will provide only clean Secret Bindings. All the dirty ones will be cleaned up regularly as before
- Subscription Cleanup Job will run twice per day

There are possible drawbacks to such a solution. Subscription Cleanup Job cleans everything up once per day now. If a customer provisions and deprovisions their SKR multiple times, all of existing Secret Bindings could be used and marked as "dirty" so there could be a situation when there is no clean one left. In this case, KEB will return an error saying `failed to find unassigned secret binding` and there will be no option to provision a new SKR till the next cleanup. To minimize such possibility I configured the Subscription Cleanup Job to run twice per day instead of only once

### CODE REVIEW

**Review by koala7659 — APPROVED** [review]
LGTM

File: components/kyma-environment-broker/common/hyperscaler/account_pool_test.go [hunk 1]
```diff
@@ -277,8 +280,36 @@ func newTestAccountPool() AccountPool {
...
+			Name:      "secret8",
+			Namespace: testNamespace,
+		},
+	}
+	secretBinding9 := &gardener_types.SecretBinding{
+		ObjectMeta: machineryv1.ObjectMeta{
+			Name:      "secretBinding9",
+			Namespace: testNamespace,
+			Labels: map[string]string{
+				"hyperscalerType": "azure",
```
[the remark this answers is not in the document]
Comment (franpog859) [reply components/kyma-environment-broker/common/hyperscaler/account_pool_test.go:303]: In the current implementation, a Secret Binding can have a `dirty` label set to `true` or just no `dirty` label. Subscription Cleanup Job [deletes the label](https://github.com/kyma-project/control-plane/blob/main/components/subscription-cleanup-job/internal/job/cleaner.go#L105) when it's not dirty anymore. In this case there is no option to have `dirty` label set to `false`

**Comment by vvxxvvxx** [comment]
lgtm

File: resources/kcp/charts/kyma-environment-broker/templates/subscription-cleanup-job.yaml [hunk 2]
```diff
@@ -30,5 +30,5 @@ spec:
             - name: gardener-kubeconfig
               secret:
                 secretName: {{ .Values.gardener.secretName }}
-  schedule: '0 2 * * *'
+  schedule: '0 2,14 * * *'
```
Comment (koala7659) [inline resources/kcp/charts/kyma-environment-broker/templates/subscription-cleanup-job.yaml:33]: It implies to run subscription-cleanup-job every 12 hours which is IMHO too rare. You should restore previous schedule for this job after problem is fixed.

**Review by koala7659 — APPROVED** [review]

File: components/kyma-environment-broker/common/hyperscaler/account_pool.go [hunk 3]
```diff
@@ -109,7 +109,7 @@ func (p *secretBindingsAccountPool) IsSecretBindingUsed(hyperscalerType Type, te
 }
 
 func (p *secretBindingsAccountPool) CredentialsSecretBinding(hyperscalerType Type, tenantName string) (*v1beta1.SecretBinding, error) {
-	labelSelector := fmt.Sprintf("tenantName=%s,hyperscalerType=%s", tenantName, hyperscalerType)
+	labelSelector := fmt.Sprintf("tenantName=%s, hyperscalerType=%s, !dirty", tenantName, hyperscalerType)
```
Comment (koala7659) [inline components/kyma-environment-broker/common/hyperscaler/account_pool.go:112]: After additional review I think we should make in both cases (L112 and L 126) same label selector. `shared!=true, tenantName=%s, hyperscalerType=%s, !dirty` . I'm not sure if we can add `internal=true` label to selector query.

**Review by koala7659 — CHANGES_REQUESTED** [review]

File: resources/kcp/charts/kyma-environment-broker/templates/subscription-cleanup-job.yaml (hunk 2 above)
Comment (piotrmiskiewicz) [reply resources/kcp/charts/kyma-environment-broker/templates/subscription-cleanup-job.yaml:33]: just a question: I'm not sure how statistics looks like, how is it possible to have all subscriptions dirty, but why not run such job every hour?

**Review by piotrmiskiewicz — APPROVED** [review]

File: components/kyma-environment-broker/common/hyperscaler/account_pool.go (hunk 3 above)
Comment (koala7659) [reply components/kyma-environment-broker/common/hyperscaler/account_pool.go:112]: Resolved!

**Review by koala7659 — APPROVED** [review]
LGTM

**Review by piotrmiskiewicz — APPROVED** [review]

**Pull request merged**